storage/innobase/row/row0umod.cc

```diff
@@ -470,6 +470,144 @@ row_undo_mod_clust(
 	return(err);
 }
 
+/** Find out if an accessible version of a clustered index record
+corresponds to a secondary index entry.
+@param rec    record in a latched clustered index page
+@param index  secondary index
+@param ientry secondary index entry
+@param mtr    mini-transaction
+@return whether an accessible non-dete-marked version of rec
+corresponds to ientry */
+static bool row_undo_mod_sec_unsafe(const rec_t *rec, dict_index_t *index,
+                                    const dtuple_t *ientry, mtr_t *mtr)
+{
```

Comment:
It may actually be required, but it is not possible at the moment. In fact there is a bug [MDEV-29823](https://jira.mariadb.org/browse/MDEV-29823) where we will leak a secondary index record indefinitely here.